src/main/java/net/minecraftforge/network/HandshakeMessages.java

```diff
@@ -68,7 +69,7 @@ public int getAsInt() {
 
         public S2CModList()
         {
-            this.mods = ModList.get().getMods().stream().map(IModInfo::getModId).collect(Collectors.toList());
+            this.mods = ModList.get().getMods().stream().map(mod -> mod.getModId()).collect(Collectors.toList());
```

Comment:
Why?

Reply:
That's a remnant of the old way of sending the mod version via a `#` as a separator, will fix